python/pyspark/sql/profiler.py

```diff
@@ -0,0 +1,176 @@
+#
```

Comment:
qq do we want to expose any of them in this file as an API?

Reply:
No, the new config and `spark.showPerfProfiles` should be the new user facing API.